crates/router/src/connector/paypal/transformers.rs

```diff
@@ -0,0 +1,604 @@
+            }),
+            ..item.data
+        })
+    }
+}
+
+#[derive(Debug)]
+pub struct PaypalAuthType {
+    pub(super) api_key: String,
+    pub(super) key1: String,
+}
+
+impl TryFrom<&types::ConnectorAuthType> for PaypalAuthType {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
+        match auth_type {
+            types::ConnectorAuthType::BodyKey { api_key, key1 } => Ok(Self {
+                api_key: api_key.to_string(),
+                key1: key1.to_string(),
+            }),
+            _ => Err(errors::ConnectorError::FailedToObtainAuthType)?,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum PaypalOrderStatus {
+    Completed,
+    Voided,
+    Created,
+    Saved,
+    PayerActionRequired,
+    Approved,
+}
+
+impl ForeignFrom<(PaypalOrderStatus, PaypalPaymentIntent)> for storage_enums::AttemptStatus {
+    fn foreign_from(item: (PaypalOrderStatus, PaypalPaymentIntent)) -> Self {
+        match item.0 {
+            PaypalOrderStatus::Completed => {
+                if item.1 == PaypalPaymentIntent::Authorize {
+                    Self::Authorized
+                } else {
+                    Self::Charged
+                }
+            }
+            PaypalOrderStatus::Voided => Self::Failure,
+            PaypalOrderStatus::Created | PaypalOrderStatus::Saved | PaypalOrderStatus::Approved => {
+                Self::Pending
+            }
+            PaypalOrderStatus::PayerActionRequired => Self::Authorizing,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct PaymentsCollectionItem {
+    amount: OrderAmount,
+    expiration_time: Option<String>,
+    id: String,
+    final_capture: Option<bool>,
+    status: PaypalOrderStatus,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub struct PaymentsCollection {
+    authorizations: Option<Vec<PaymentsCollectionItem>>,
+    captures: Option<Vec<PaymentsCollectionItem>>,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub struct PurchaseUnitItem {
+    reference_id: String,
+    payments: PaymentsCollection,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PaypalPaymentsResponse {
+    id: String,
+    intent: PaypalPaymentIntent,
+    status: PaypalOrderStatus,
+    purchase_units: Vec<PurchaseUnitItem>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PaypalPaymentsSyncResponse {
+    id: String,
+    status: PaypalPaymentStatus,
+    amount: OrderAmount,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PaypalMeta {
+    pub authorize_id: Option<String>,
+    pub order_id: String,
+    pub psync_flow: PaypalPaymentIntent,
+}
+
+fn get_auth_or_capt(
+    intent: PaypalPaymentIntent,
+    purchase_unit: &PurchaseUnitItem,
+) -> CustomResult<String, errors::ConnectorError> {
+    || -> _ {
```

Comment:
Why is the closure being used here? It can be done directly right? and if this block always returns `Some` value then it is not required to return it as an option

Reply:
Previously, I was handling the errors in three places and now it is used to handle all the errors in one place.